test/raft/randomized_nemesis_test.cc

```diff
@@ -2179,6 +2193,84 @@ SEASTAR_TEST_CASE(basic_test) {
     tlogger.debug("Finished");
 }
 
+SEASTAR_TEST_CASE(test_frequent_snapshotting) {
+    auto seed = tests::random::get_int<int32_t>();
+    std::mt19937 random_engine{seed};
+
+    logical_timer timer;
+    environment_config cfg {
+        .rnd{random_engine},
+        .network_delay{0, 6},
+        .fd_convict_threshold = 50_t,
+    };
+    co_await with_env_and_ticker<ExReg>(cfg, [&timer] (environment<ExReg>& env, ticker& t) -> future<> {
+        using output_t = typename ExReg::output_t;
+
+        t.start([&] (uint64_t tick) -> future<> {
+            env.tick_network();
+            timer.tick();
+            if (tick % 10 == 0) {
+                env.tick_servers();
+            }
+            return ping_shards();
+        }, 10'000);
+        const auto server_config = raft::server::configuration {
+            .snapshot_threshold{1},
+            .snapshot_trailing{5},
+            .max_log_size{20},
+            .enable_forwarding{true}
+        };
+
+        auto leader_id = co_await env.new_server(true, server_config);
+
+        auto call = [&] (ExReg::input_t input, raft::logical_clock::duration timeout) {
+            return env.call(leader_id, std::move(input),  timer.now() + timeout, timer);
+        };
+
+        auto eq = [] (const call_result_t<ExReg>& r, const output_t& expected) {
+            return std::holds_alternative<output_t>(r) && std::get<output_t>(r) == expected;
+        };
+
+        // Wait at most 1000 ticks for the server to elect itself as a leader.
+        assert(co_await wait_for_leader<ExReg>{}(env, {leader_id}, timer, timer.now() + 1000_t) == leader_id);
+
+        auto id2 = co_await env.new_server(false, server_config);
+        auto id3 = co_await env.new_server(false, server_config);
+
+        env.for_each_server([](raft::server_id, raft_server<ExReg>* srv) {
+            srv->get_server()->set_applier_queue_max_size(1);
```

Comment:
Could've done this change in `basic_generator_test` (in a random part of the runs), would cover more ground (the test is running nemeses etc.)

Reply:
I like the separate test more. It is impossible to account for all possible scenarios in one test, and the more such scenarios we add to it, the harder it becomes to maintain and debug.

Reply:
We don't have to choose, we can do both.

The purpose of `basic_generator_test` is to test different combinations of parameters, not necessarily in the same run but across different runs (you can then reproduce a combination using a seed), so given enough runs all combinations will eventually be tested.

We're now not testing what happens e.g. when the queue size is 1 and we're crashing servers or doing network partitions, while we easily could, without writing 100 more tests but by adding a little piece of code to the beginning of `basic_generator_test`.

Reply:
`basic_generator_test` catched tens of bugs when I first wrote it. The chaos is what makes it so effective.

Reply:
Yes, I agree. We could sometimes randomly set applier queue size to 1 in `basic_generator_test`. These are just different kinds of tests, `basic_generator_test` is 'exploratory', intended to find bugs we don't know about yet, and this test is 'functional', intended to perpetuate that certain functionality.